web_query_timeout/controllers/main.py

```diff
@@ -25,3 +25,21 @@ def do_search_read(self, model, fields=False, offset=0, limit=False,
                 'length': 0,
                 'records': 'WebQueryTimeoutException',
             }
+
+    def _call_kw(self, model, method, args, kwargs):
+        """
+        Set the context value that triggers the database query duration when
+        the triggered method is read_group.
+        """
+        if method == 'read_group':
```

Comment:
This is a very generic place to put it, and it also covers xmlrpc usage which is less desirable. I think you can set the context here: https://github.com/odoo/odoo/blob/8.0/addons/web/static/src/js/data.js#L40 (and that would also replace line 17 of this file).